recipes/tls-securepair-to-tlssocket/src/workflow.ts

```diff
@@ -0,0 +1,114 @@
+import { getNodeImportStatements, getNodeImportCalls as getNodeNodeImportCall } from "../../../utils/src/ast-grep/import-statement.ts";
+import { getNodeRequireCalls } from "../../../utils/src/ast-grep/require-call.ts";
+import { updateBinding } from "../../../utils/src/ast-grep/update-binding.ts";
+import { removeLines } from "../../../utils/src/ast-grep/remove-lines.ts";
+import type { Edit, SgRoot, Range, SgNode } from "@codemod.com/jssg-types/main";
+import type Js from "@codemod.com/jssg-types/langs/javascript";
+
+function getClosest(node: SgNode<Js>, kinds: string[]): SgNode<Js> | null {
+  let current = node.parent();
+ 
+  while (current) {
+    if (kinds.includes(current.kind())) return current;
+    current = current.parent();
+  }
+ 
+  return null;
+}
+
+export default function transform(root: SgRoot<Js>): string | null {
+  const rootNode = root.root();
+  const edits: Edit[] = [];
+  const linesToRemove: Range[] = [];
+
+  const importNodes = [
+    ...getNodeImportStatements(root, "tls"),
+    ...getNodeRequireCalls(root, "tls"),
+    ...getNodeNodeImportCall(root, "tls")
+  ];
+
+  for (const node of importNodes) {
+    const change = updateBinding(node, { old: "SecurePair", new: "TLSSocket" });
+    if (change?.edit) edits.push(change.edit);
+    if (change?.lineToRemove) linesToRemove.push(change.lineToRemove);
+  }
+
+  const newExpressions = rootNode.findAll({
+    rule: {
+      kind: "new_expression",
+      has: {
+        any: [
+          { kind: "member_expression", has: { field: "property", regex: "^SecurePair$" } },
+          { kind: "identifier", regex: "^SecurePair$" }
```

Comment:
nit: The match can be simplified by using a pattern instead of a regex.
```suggestion
          { kind: "identifier", pattern: "SecurePair" }
```